### PR TITLE
Fix dashboard editing permissions not working

### LINKED ISSUE #4612: Dashboard editing permissions not working
### Issue Summary

Suppose user **A** shares dashboard edit permissions with user **B**. When user **B** visits the dashboard, there is no edit button. User **B** also can't make a `POST` api call with changes.

### Steps to Reproduce

1. Ensure the permissions feature flag is ticked.
2. Share dashboard edit permissions with another user
3. Observe the other user does not have an edit button when visiting the dashboard.

### Technical details:

* Redash Version: V9 
* Browser/OS: FF & API client
* How did you install Redash: SaaS

### PR DESCRIPTION
**Pull request by gabrieldutra**
## What type of PR is this? (check all applicable)

- [x] Bug Fix

## Description
This updates the check for `canEditDashboard` to use the resource attribute `can_edit`.

## Related Tickets & Documents
Fixes #4612 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
--

### CODE REVIEW

File: client/app/pages/dashboards/hooks/useDashboard.js [hunk 1]
```diff
@@ -37,7 +37,8 @@ function useDashboard(dashboardData) {
   const [refreshing, setRefreshing] = useState(false);
   const [gridDisabled, setGridDisabled] = useState(false);
   const globalParameters = useMemo(() => dashboard.getParametersDefs(), [dashboard]);
-  const canEditDashboard = useMemo(
+  const canEditDashboard = useMemo(() => !dashboard.is_archived && dashboard.can_edit, [dashboard]);
```
Comment (arikfr) [inline client/app/pages/dashboards/hooks/useDashboard.js]: Why use `useMemo` here? This is just assignment of two booleans.

**Comment by arikfr** [comment]
@susodapop can you confirm this is fixed?

**Comment by susodapop** [comment]
Yes, fixed 👍

**Comment by arikfr** [comment]
@gabrieldutra can you make the same fix on the SaaS master branch? Thanks.

**Pull request merged**